tests/release-build.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# test name and purpose
+echo ''
+echo '                         ##### Release Build Test #####'
+echo ''
+echo '    The purpose of this test is to ensure that nodeos was built with compiler'
```

Comment:
To add to @swatanabe-b1's comment [here](https://github.com/EOSIO/eos/pull/7882#issuecomment-530543223),  this test isn't truly checking if nodeos was built with debug symbols nor is it checking which optimization level it was built with. It is, however, checking part of the build environment (specifically the debug flag) embedded in nodeos which determines whether nodeos is compatible with a particular Chainbase file it is attempting to open. Checking whether the released binaries are compatible with existing Chainbase files is probably the more important thing to check as part of our CICD process; the CICD sync tests do a good job of that but this test is intended to be a faster way of catching a class of problems that we know would lead to failures in our sync tests later on. However, due to correlation between the optimization setting and status of `NDEBUG`, this test also kind of acts as a check of the compiler optimization used (at least for the typical way the build would be configured with `CMAKE_BUILD_TYPE`).